src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
Is it really tied to the client rather than a session? The sync client created by the object store lives indefinitely. One sync client services multiple sessions.